app/src/org/commcare/gis/EntityMapActivity.java

```diff
@@ -368,6 +401,50 @@ private void setMapLocationEnabled(boolean enabled) {
         }
     }
 
+    private void setupMapToggles() {
+        View overlay = findViewById(R.id.map_overlay);
+
+        int visibleToggles = 0;
+
+        if (numMarkers > 1) {
+            toggleMarkers.setVisibility(View.VISIBLE);
+            visibleToggles++;
+            toggleMarkers.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                for (Marker marker : markerReferences.keySet()) {
+                    marker.setVisible(isChecked);
+                }
+            });
+        } else {
+            toggleMarkers.setVisibility(View.GONE);
+        }
+
+        if (numPolygons > 1) {
+            togglePolygons.setVisibility(View.VISIBLE);
+            visibleToggles++;
+            togglePolygons.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                for (Polygon polygon : polygonInfo.keySet()) {
+                    polygon.setVisible(isChecked);
+                }
+            });
+        } else {
+            togglePolygons.setVisibility(View.GONE);
+        }
+
+        if (numGeoPoints > 1) {
+            toggleGeoPoints.setVisibility(View.VISIBLE);
+            visibleToggles++;
+            toggleGeoPoints.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                for (Circle circle : geoPointCircles) {
+                    circle.setVisible(isChecked);

```

Comment:
Curious if we have checked how does it perform for thousands of markers, does it result into all of them getting disappeated together vs a few at a times.